### DISCUSSION TITLE
When playing two MP4 videos at the same time

### DISCUSSION THREAD

**Discussion by koreaflash** [airsdk/Adobe-Runtime-Support#1626] [General]
When playing two MP4 videos at the same time, one plays while the other generates a "NetStream.Play.Failed" error.
- It only occurs on Android and does not occur on MS Windows.
- Both FLV videos are played normally.
- If one is FLV and the other is MP4, both are played normally.
- In versions prior to AIR 32.0.0.116, both play normally.

Why?

**Comment by ajwfrost** [comment]
On Android we have one stream of MP4 support available from the OS, so the first one ties up this capability. In earlier AIR versions, any second stream was then decoded using software, but the patent licensing for this is no longer valid so these are now failing.

I am wondering whether that limitation of only one hardware-decoded video is still a problem on today's devices so we can check to see if there's a way to update this and allow multiple videos simultaneously on Android. As you've seen, on Windows there can be multiple videos and these are being handled by the OS..  we'll investigate further on this.

thanks

   Andrew

**Reply by koreaflash** [reply]
We use AIR App to provide educational content.
It is provided in units of pages, and the next page is preloaded for smooth progress.
Problems may arise here. If there is video content on the current page and video content on the next page, the video content on the next page cannot be played.

**Reply by lwmirkk** [reply]
stop one video before play the next one is not an option? you can do this pretty simple with MediaPlayer ANE from distriqt (https://airnativeextensions.com/extension/com.distriqt.MediaPlayer)

**Reply by koreaflash** [reply]
Thanks for your comments.
I'll consider it.

**Reply by itlancer** [reply]
> stop one video before play the next one is not an option?

In many cases this is not an option cause for smooth UX you need at least stop previous video, make it screenshot (`BitmapData`), display it, start to play new video and remove screenshot.
But with AIR for Android it cannot be done right now. Here bug report: https://github.com/airsdk/Adobe-Runtime-Support/issues/87
So right now seems only one way to achieve it - use native extensions.
